### DISCUSSION TITLE
Is there a reset control API in Zephyr?

### DISCUSSION THREAD

**Discussion by billy-tsai** [zephyrproject-rtos/zephyr#38062] [Q&A]
Hi, 
I want to know that if the zephyr has a common API like Linux: [reset](https://github.com/torvalds/linux/blob/master/include/linux/reset.h) to control the reset?

**Comment by dlktdr** [comment]
It has a sys_reboot function,

sys_reboot(SYS_REBOOT_COLD);

You will need the CONFIG_REBOOT=y KConfig option set to be able to use it. https://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_REBOOT.html

**Reply by billy-tsai** [reply]
Hi @dlktdr,
Thanks for your reply.
But my question is common API that can be used in the driver init function, like reset_control_deassert(...) can deassert the reset of the device, not the function used to reboot the system.

**Comment by yashi** [comment]
I don't think there is a unified API for reset pins. 

The current (v2.7) standard way is to use GPIO,
But pinctrl discussion is on going (#35077)
